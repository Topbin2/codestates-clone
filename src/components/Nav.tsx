import styled from "styled-components";
import black_svg from "../assets/icon/codestates2.png";
import { GiHamburgerMenu } from "react-icons/gi";
import { IoPersonCircleSharp } from "react-icons/io5";
import { BREAK_POINT_TABLET } from "../constants/mediaQuery";
import ProfileDropDown from "./ProfileDropDown";
import { useAppDispatch } from "../hooks/reduxHooks";
import { toggleProfileDropDownOpen } from "../reducers/toggleSlice";
import { Link } from "react-router-dom";

const Wrapper = styled.header`
  width: 100%;
  height: 65px;

  .contentsContainer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 24px;
    box-shadow: rgb(0 0 0 / 20%) 0px 0px 4px 0px;

    .logoContainer {
      height: 100%;
      padding: 12px;
      float: left;
      .logo_icon {
        margin-top: 5px;
        height: 24px;
        cursor: pointer;
      }
    }
  }

  .menuIcon {
    width: 24px;
    height: 24px;
    color: rgb(82, 92, 101);
    margin: 2px 5px;
  }

  .tabContainer {
    display: none;
    position: relative;
  }

  @media only screen and (min-width: ${BREAK_POINT_TABLET}px) {
    .menuIcon {
      display: none;
    }

    .tabContainer {
      display: flex;
      align-items: center;

      button {
        padding: 10px;
        color: ${(props) => props.theme.textColor};
        margin-right: 20px;
        font-size: 15px;
        font-weight: 600;
        border: 0;
        border-radius: 10px;
        outline: 0;
        background-color: ${(props) => props.theme.backgroundColor};
        cursor: pointer;

        &:hover {
          background-color: ${(props) => props.theme.hoverBackgroundColor};
        }
      }

      .jobs {
        font-size: 16px;
      }

      .loginBtn {
        width: 30px;
        height: 30px;
        color: ${(props) => props.theme.textColor};
        cursor: pointer;
      }
    }
  }
`;

const Nav = () => {
  const dispatch = useAppDispatch();
  return (
    <Wrapper>
      <div className="contentsContainer">
        <div className="logoContainer">
          <Link to="/">
            <img src={black_svg} className="logo_icon" alt="logo_icon" />
          </Link>
        </div>
        <nav>
          <GiHamburgerMenu className="menuIcon" />
          <div className="tabContainer">
            <Link to="/course/active" data-testid="link1">
              <button className="tab">수강 중인 코스</button>
            </Link>
            <Link to="/coplit" data-testid="link2">
              <button>코플릿</button>
            </Link>
            <Link to="/job" data-testid="link3">
              <button className="jobs">Jobs</button>
            </Link>
            <IoPersonCircleSharp
              className="loginBtn"
              onMouseOver={() => dispatch(toggleProfileDropDownOpen())}
            />
            <ProfileDropDown />
          </div>
        </nav>
      </div>
    </Wrapper>
  );
};

export default Nav;
