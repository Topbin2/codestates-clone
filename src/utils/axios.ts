import axios from "axios";

export const custom = axios.create({
  baseURL: "https://jsonplaceholder.typicode.com",
});
