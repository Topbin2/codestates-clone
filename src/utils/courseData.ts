export const courseData = [
  {
    id: 1,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Section 1",
    body: "웹 프론트엔드 개발의 기초지식을 기반으로 스스로 단순한 Web App을 만들 수 있다.",
  },
  {
    id: 2,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Section 2",
    body: "서버와 통신이 가능한 구조적인 Web App을 만들 수 있다.",
  },
  {
    id: 3,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Section 3",
    body: "사용자 친화적이고 안전한 Web App을 만들 수 있다.",
  },
  {
    id: 4,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Section 4",
    body: "사람과 기계가 모두 쉽고 빠르게 접근 가능한 Web App을 만들 수 있다.",
  },
  {
    id: 5,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Project",
    body: "First Project, Final Project",
  },
  {
    id: 6,
    url: "https://urclass.codestates.com/static/media/img-cover.23e06f01.jpg",
    title: "[SEB FE] Wiki & Survey",
    body: "SEB FE 부트캠프에서 활용되고 있는 Wiki & Survey",
  },
];
