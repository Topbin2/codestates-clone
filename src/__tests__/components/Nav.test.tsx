import Nav from "../../components/Nav";
import { render, screen } from "../../utils/test-utils";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import setupStore from "../../store/store";
import { BrowserRouter as Router } from "react-router-dom";

describe("Nav", () => {
  it("snapshot", () => {
    const component = renderer.create(
      <Provider store={setupStore()}>
        <Router>
          <Nav />
        </Router>
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("nav menu 버튼은 해당 속성을 포함한다.", () => {
    render(<Nav />);
    const logo = screen.getByRole("link");
    const tab1 = screen.getByTestId("link1");
    const tab2 = screen.getByTestId("link2");
    const tab3 = screen.getByTestId("link3");

    expect(logo).toHaveAttribute("href", "/");
    expect(tab1).toHaveAttribute("href", "/course/active");
    expect(tab2).toHaveAttribute("href", "/coplit");
    expect(tab3).toHaveAttribute("href", "/job");
  });
});
