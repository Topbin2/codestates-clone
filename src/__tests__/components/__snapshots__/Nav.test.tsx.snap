// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot 1`] = `
<header
  className="sc-gsnTZi dgJSci"
>
  <div
    className="contentsContainer"
  >
    <div
      className="logoContainer"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo_icon"
          className="logo_icon"
          src="codestates2.png"
        />
      </a>
    </div>
    <nav>
      <svg
        className="menuIcon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
        />
      </svg>
      <div
        className="tabContainer"
      >
        <a
          data-testid="link1"
          href="/course/active"
          onClick={[Function]}
        >
          <button
            className="tab"
          >
            수강 중인 코스
          </button>
        </a>
        <a
          data-testid="link2"
          href="/coplit"
          onClick={[Function]}
        >
          <button>
            코플릿
          </button>
        </a>
        <a
          data-testid="link3"
          href="/job"
          onClick={[Function]}
        >
          <button
            className="jobs"
          >
            Jobs
          </button>
        </a>
        <svg
          className="loginBtn"
          fill="currentColor"
          height="1em"
          onMouseOver={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm2 96a72 72 0 11-72 72 72 72 0 0172-72zm-2 288a175.55 175.55 0 01-129.18-56.6C135.66 329.62 215.06 320 256 320s120.34 9.62 129.18 55.39A175.52 175.52 0 01256 432z"
          />
        </svg>
        <div
          className="sc-bczRLJ bOLUDk"
          onMouseLeave={[Function]}
        >
          <div
            className="me"
          >
            <span
              className="name"
            >
              모상빈
            </span>
            님, 안녕하세요!
          </div>
          <div
            className="profileText"
          >
            <svg
              className="profileIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M19 14v3c0 .55-.45 1-1 1h-1v-4h2M7 14v4H6c-.55 0-1-.45-1-1v-3h2m5-13a9 9 0 00-9 9v7c0 1.66 1.34 3 3 3h3v-8H5v-2c0-3.87 3.13-7 7-7s7 3.13 7 7v2h-4v8h3c1.66 0 3-1.34 3-3v-7a9 9 0 00-9-9z"
              />
            </svg>
             
            <span>
              수강생 헬프센터
            </span>
          </div>
          <div
            className="profileText"
          >
            <svg
              className="profileIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 8a3 3 0 1 0 0-6 3 3 0 0 0 0 6zm2-3a2 2 0 1 1-4 0 2 2 0 0 1 4 0zm4 8c0 1-1 1-1 1H3s-1 0-1-1 1-4 6-4 6 3 6 4zm-1-.004c-.001-.246-.154-.986-.832-1.664C11.516 10.68 10.289 10 8 10c-2.29 0-3.516.68-4.168 1.332-.678.678-.83 1.418-.832 1.664h10z"
              />
            </svg>
            <span>
              마이페이지
            </span>
          </div>
          <div
            className="profileText logout"
          >
            <svg
              className="profileIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M17 7l-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
              />
            </svg>
            <span>
              로그아웃
            </span>
          </div>
        </div>
      </div>
    </nav>
  </div>
</header>
`;
